### PR TITLE
Remove OrchardCore.OpenId.EntityFrameworkCore and OrchardCore.OpenId.Core.EntityFrameworkCore

### PR DESCRIPTION
**Pull request by PinpointTownes**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/1982.

### CODE REVIEW

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Resolvers/OpenIdApplicationStoreResolver.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            {
+                return store;
+            }
+
+            var type = _cache.GetOrAdd(typeof(TApplication), key =>
+            {
+                if (!typeof(OpenIdApplication).IsAssignableFrom(key))
+                {
+                    throw new InvalidOperationException(new StringBuilder()
+                        .AppendLine("The specified application type is not compatible with the YesSql stores.")
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Resolvers/OpenIdApplicationStoreResolver.cs]: Shouldn't it be a const string instead?
Perf is not a problem here but it looks odd.

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Resolvers/OpenIdScopeStoreResolver.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+using OrchardCore.OpenId.YesSql.Stores;
+
+namespace OrchardCore.OpenId.YesSql.Resolvers
+{
+    /// <summary>
+    /// Exposes a method allowing to resolve a scope store.
+    /// </summary>
+    public class OpenIdScopeStoreResolver : IOpenIddictScopeStoreResolver
+    {
+        private readonly ConcurrentDictionary<Type, Type> _cache = new ConcurrentDictionary<Type, Type>();
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Resolvers/OpenIdScopeStoreResolver.cs]: Cache of types, can it be static of does it depend on specific tenant configuration?

Comment (PinpointTownes) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Resolvers/OpenIdScopeStoreResolver.cs]: Nope, but I updated the resolvers to use private singleton caches injected via DI (it's the same approach as the one used in OpenIddict itself). See https://github.com/openiddict/openiddict-core/issues/726 for more context.

File: src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Resolvers/OpenIdApplicationStoreResolver.cs (hunk 1 above)
Comment (PinpointTownes) [reply src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Resolvers/OpenIdApplicationStoreResolver.cs]: It's a pattern I frequently use as it's a quite convenient way to create line breaks in exception messages (without having to use `Environment.NewLine`). Feel free to change that if you don't like it.

**Comment by sebastienros** [comment]
image


It's called accountability

**Comment by PinpointTownes** [comment]
Hey, I'm a serious guy... I waited for the CI builds to be green before merging it :trollface:

**Comment by PinpointTownes** [comment]
When shipping 1.0.0, you'll probably want to unlist these 2 packages so they don't appear in NuGet's search results.

**Comment by sebastienros** [comment]
Practically speaking, most of our packages (not modules) might be unlisted, as they should probably not be referenced directly. Is that a think in Nuget metadata to publish a package and not have it indexed for search, just for dependencies?

**Comment by PinpointTownes** [comment]
I don't think packages authors should unlist packages for the sole reason they are not meant to be "directly" referenced.

In this specific case, I suggested unlisting both the OpenID EF Core "core" and "module" packages because they'll no longer be produced nor supported and so shouldn't be used (at all, directly or indirectly).

**Pull request merged**